### ISSUE TITLE
Error when trying to sync the project config: Changes to the project config are not possible while in read-only mode.

### ISSUE THREAD

**Issue by smcyr** [craftcms/cms#3957]
### Description

I updated my website in local (which updated my project.yaml) and uploaded it to my staging. On staging, I tried to run the project-config/sync command, but it gave me the error "Changes to the project config are not possible while in read-only mode." When I debug, I can see that the culprit is a matrix field.

I could send you my database (and project.yaml), but it's 200MB.

### Additional info

- Craft version: 3.1.16
- PHP version: 7.2
- Database driver & version: MariaDB

**Comment by smcyr** [comment]
Update: if I run the update with `allowAdminChanges` to `true`, the `project.yaml` is updated with all the boolean (`true`, `false`) updated to int (`0`, `1`). I don't think it should do that if you're just syncing the project config.
Thanks!

**Comment by brandonkelly** [comment]
@smcyr Any chance you can send your incoming `project.yaml` file, as long as a database backup from before the sync, and your `composer.lock` & `composer.json` files over to support@craftcms.com?

**Comment by smcyr** [comment]
@brandonkelly I sent you an email with everything. Thanks and have a good weekend!

**Comment by cholawo** [comment]
I've seen this issue - it was caused by using two different versions of php, I had one running 7.1 and the read-only one running 7.2.1.   The read-only install attempts to change the booleans and then gets stuck.   Once I made my craft installs use the same php version this issue was solved for me.

**Comment by smcyr** [comment]
Some of our environments are PHP 7.1 and others PHP 7.2 also, so maybe it's the cause. 
But it should be made to work with different PHP versions in my opinion.

**Comment by cholawo** [comment]
Yes I agree of course it should work - this is flaky behaviour.   Since my previous comment I am actually still experiencing this issue, but I think at least it's not caused by different php versions now - there must be more reasons as to why it's happening.   This doesn't fill me with confidence in the project config file, whilst I had such high hopes for it before experiencing these issues.    It seems to me that it starts to crop up more as the number and complexity of fields increases.

**Comment by smcyr** [comment]
In my case, the website is definitely complex, it's over 200MB. And we are multiple people working on the project, making changes to the project config. Maybe it's because of the database versions too. I know that some have MySQL and other MariaDB. 
Hoping that all edge cases are resolved over time.

**Comment by nettum** [comment]
Possibly related to https://github.com/craftcms/cms/issues/3695 ?

I'm trying to find a reliable way to use `project.yaml` with `./craft migrate/all` and `./craft project-config/sync` on deployment.

On the latest deployment I got `> renaming project.yaml to project.yaml.J8eJvNK3oT ... Exception: Changes to the project config are not possible while in read-only mode.`

The env I deploy to got `'allowAdminChanges' => false`.

**Comment by brandonkelly** [comment]
@nettum You should pick a single environment to run the actual 3.1 migration on (ideally on a local environment with a fresh backup from production), and then create a new DB backup from that environment, and restore across the others. The main 3.1 migration **does not support** reading an existing project.yaml file.

**Comment by nettum** [comment]
Thanks, Brandon. That's the workflow I'm using apart from dumping the db back to stage/prod env. Will do that after the initial 3.1 upgrade on local dev.

**Comment by mdominguez** [comment]
Experiencing this issue of @smcyr too, my local environment runs php 7.3.1 and production 7.3.0, is there any way to work around this without matching php versions on both environments?
